src/download.jl

```diff
@@ -0,0 +1,72 @@
+using Printf
+using DataDeps
+
+# generate_dataset_url helper function
+# This function takes in a dataset code of format "GADM/<country_code>"
+# It returns the URL of the gpkg dataset of the country
+
+function generate_dataset_url(dataset_code)
+
+    # Splits GADM/IND to GADM, IND
+    dataset_provider, country_code = split(dataset_code, "/")
+    
+    # Only accepts GADM Dataset Provider
+    if dataset_provider != "GADM"
+        println("❌ Dataset Provider $dataset_provider not supported.\n")
+        println("💡 Please try \"GADM/<Code>\".")
+        return ""
+    end
+    
+    println("✅ fetching $country_code's data from GADM\n")
+
+    dataset_url = @sprintf "https://biogeo.ucdavis.edu/data/gadm3.6/gpkg/gadm36_%s_gpkg.zip" country_code
```

Comment:
Julia has string interpolation, so no need to use sprintf. You can just add `$variable` inside the string to get the contents in place. Also, this would remove the necessity of the `Printf` dependency.